test/cases/enum.ts

```diff
@@ -20,10 +21,158 @@ export var schema = {
   "additionalProperties": false
 }
 
-export var types = `interface Enum {
+export var configurations = [
+  {
+    settings: {
+      useTypescriptEnums: false
+    },
+    types: `interface Enum {
   foo: "a" | "b" | "c";
   bar: number;
   baz: {
     a: number;
   };
 }`
+  }, 
+  {
+    settings: {
+      useTypescriptEnums: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+enum Bar {
+  One = 1,
+  Two = 2,
+  Three = 3
+}
+enum Baz {
+  [object Object],
+  [object Object],
+  [object Object]
+}
+interface Enum {
+  foo: Foo;
+  bar: Bar;
+  baz: Baz;
+}`
+  },
+    {
+    settings: {
+      useTypescriptEnums: true,
+      addEnumUtils: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+class FooUtil {
```

Comment:
I think runtime checks are a powerful idea - I also want to be careful that they are designed in a way that will scale to other checks supported by JSON-Schema (see https://github.com/bcherny/json-schema-to-typescript/#not-expressible-in-typescript).

See my comment [above](https://github.com/bcherny/json-schema-to-typescript/pull/15#discussion_r77439735) - maybe we can brainstorm to see if there is another way to express what you need, without resorting to runtime checks.